openpype/tools/ayon_utils/widgets/folders_widget.py

```diff
@@ -323,16 +361,116 @@ def refresh(self):
 
         self._folders_model.refresh()
 
-    def _on_project_selection_change(self, event):
-        project_name = event["project_name"]
-        self._set_project_name(project_name)
+    def get_project_name(self):
+        """Project name in which folders widget currently is.
+
+        Returns:
+            Union[str, None]: Currently used project name.
+        """
+
+        return self._folders_model.get_project_name()
+
+    def set_project_name(self, project_name):
+        """Set project name.
+
+        Do not use this method when controller is handling selection of
+        project using 'selection.project.changed' event.
+
+        Args:
+            project_name (str): Project name.
+        """
 
-    def _set_project_name(self, project_name):
         self._folders_model.set_project_name(project_name)
 
+    def get_selected_folder_id(self):
+        """Get selected folder id.
+
+        Returns:
+            Union[str, None]: Folder id which is selected.
+        """
+
+        return self._get_selected_item_id()
+
+    def get_selected_folder_label(self):
+        """Selected folder label.
+
+        Returns:
+            Union[str, None]: Selected folder label.
+        """
+
+        item_id = self._get_selected_item_id()
+        return self.get_folder_label(item_id)
+
+    def get_folder_label(self, folder_id):
+        """Folder label for a given folder id.
+
+        Returns:
+            Union[str, None]: Folder label.
+        """
+
+        index = self._folders_model.get_index_by_id(folder_id)
+        if index.isValid():
+            return index.data(QtCore.Qt.DisplayRole)
+        return None
+
+    def set_selected_folder(self, folder_id):
+        """Change selection.
+
+        Args:
+            folder_id (Union[str, None]): Folder id or None to deselect.
+        """
+
+        if folder_id is None:
+            self._folders_view.clearSelection()
+            return True
+
+        if folder_id == self._get_selected_item_id():
+            return True
+        index = self._folders_model.get_index_by_id(folder_id)
+        if not index.isValid():
+            return False
+
+        proxy_index = self._folders_proxy_model.mapFromSource(index)
+        if not proxy_index.isValid():
+            return False
+
+        selection_model = self._folders_view.selectionModel()
+        selection_model.setCurrentIndex(
+            proxy_index, QtCore.QItemSelectionModel.SelectCurrent
+        )
+        return True
+
+    def set_deselectable(self, enabled):
+        """Set deselectable mode.
+
+        Items in view can be deselected.
+
+        Args:
+            enabled (bool): Enable deselectable mode.
+        """
+
+        self._folders_view.set_deselectable(enabled)
+
+    def _on_left_double_click(self):
+        self.double_clicked_left.emit()
+        self.double_clicked.emit()
```

Comment:
Wouldn't this technically make more sense if it somehow would also emit along event data describing how the double click occurred? (e.g. middle-mouse?)

Reply:
Will try to use `QtCore.Qt.Mouse...` constants...

Reply:
Haha - couldn't we emit the original event data? :)

Reply:
Yes, but I'm not sure if `Signal` can handle the constants (I mean in PySide6).